tests/cpu_esp_hw_counter/README.md

```diff
@@ -0,0 +1,18 @@
+# ESP32 Periph Timer Test Using Hardware Counters
+
+## About
+
+The application corresponds exactly to the application `tests/periph_timer`.
+It is only for testing the timer implementation using hardware counters.
+In fact, the application is nothing more than the `tests/periph_timer`
+application compiled with the following command:
```

Comment:
Instead of copying every test to have a version with hardware counters, why not add a test-esp board that has
```
USEMODULE += esp_hw_counter
```

What is used by default instead? When would the user chose to use hardware counters?

Reply:
ESP32 has two counters and four 64-bit timers that can be used as `periph_timer`. By default, the 64-bit timers are used.

Reply:
> Not a single test application - just use the existing tests `tests/periph_i2c`, `tests/periph_timer` but have a board that uses the other timer / i2c implementation by default.
> At least that's how I understand the problem.

But I wouldn't like to expose this test board in the `boards` directory.

Reply:
Maybe adding an external board that lives inside the test works.
I'm not sure what the latest state of compatibility with external boards and in-tree boards was, but if that works it would probably be the cleanest solution.